ngraph/core/reference/include/ngraph/runtime/reference/fake_quantize.hpp

```diff
@@ -53,182 +56,245 @@ namespace ngraph
+                    Bound bound;
+                    std::vector<size_t> row_strides;
+                    const T* const bounds;
                 };
-                for (size_t i = 0; i < shape_size(arg_shape); ++i)
+
+                template <typename T>
+                inline T quantize(const T& arg,
+                                  const T& in_low,
+                                  const T& in_high,
+                                  const T& out_low,
+                                  const T& out_high,
+                                  const size_t& levels)
                 {
-                    T in_low_val = get_value(
-                        in_low_trivial_broadcast, in_low_aligned, in_low, i, in_low_offsets);
-                    T in_high_val = get_value(
-                        in_high_trivial_broadcast, in_high_aligned, in_high, i, in_high_offsets);
-                    T out_low_val = get_value(
-                        out_low_trivial_broadcast, out_low_aligned, out_low, i, out_low_offsets);
-                    T out_high_val = get_value(out_high_trivial_broadcast,
-                                               out_high_aligned,
-                                               out_high,
-                                               i,
-                                               out_high_offsets);
-                    if (arg[i] <= std::min(in_low_val, in_high_val))
+                    if (arg <= std::min(in_low, in_high))
+                    {
+                        return out_low;
+                    }
+                    else if (arg > std::max(in_low, in_high))
                     {
-                        out[i] = out_low_val;
+                        return out_high;
                     }
-                    else if (arg[i] > std::max(in_low_val, in_high_val))
+                    return std::nearbyint((arg - in_low) / (in_high - in_low) * (levels - 1)) /
+                               (levels - 1) * (out_high - out_low) +
+                           out_low;
+                }
+
+            } // namespace fake_quantize_details
+            namespace v1
+            {
+                template <typename T>
+                void fake_quantize(const T* const arg,
+                                   const T* const in_low,
+                                   const T* const in_high,
+                                   const T* const out_low,
+                                   const T* const out_high,
+                                   T* const out,
+                                   const Shape& arg_shape,
+                                   const Shape& in_low_shape,
+                                   const Shape& in_high_shape,
+                                   const Shape& out_low_shape,
+                                   const Shape& out_high_shape,
+                                   size_t levels,
+                                   const op::AutoBroadcastSpec& broadcast)
```

Comment:
Can we add default value for  `const op::AutoBroadcastSpec& broadcast)` looks like after that we can avoid v1 implementation.

Reply:
I will add default parameter and run CI.

Reply:
CI won't pass https://github.com/openvinotoolkit/openvino/pull/6478/commits/9d2c00d9674d0cdddac33d097ff46f527abc7bd9 changes:
```
/arm_plugin/src/arm_converter/arm_converter.hpp:302:48: error: too few arguments to function call, expected 13, have 12
[2021-07-13T06:24:11.800Z]                 _callable(std::get<I>(_args)...);
[2021-07-13T06:24:11.800Z]                 ~~~~~~~~~                      ^
```